### DISCUSSION TITLE
Cannot update database from HO 1.434

### DISCUSSION THREAD

**Discussion by radpopl** [ho-dev/HattrickOrganizer#1809] [Q&A]
I'm trying to update the database from version 1.434. I installed HO 3.0 but I get the message:
"Fatal database error. Exiting HO!
You should restore..."
Then the program asks you to indicate the database. But whatever I choose, the program freezes after selecting it ("Initialize database")
The base is correct because it works in 1.434. How to migrate to the new version?

Edit: My database is 500MB.

**Comment by wsbrenk** [comment]
Current HO version are not able to upgrade older database versions than 3.0. 

the only way to keep the 1.4 data is a temporary installation of HO3.

**Reply by radpopl** [reply]
Yes, I know, I have 3.0. And can't import from 1.434.

**Comment by wsbrenk** [comment]
I just did some research in the development history. The updateability of versions prior to 3 was removed in 2021. If the database API is still able to support the old databases today, I could try to rebuild these capabilities.
Since there still seem to be quite a few users using version 1.4, this might be a good idea. I just found an ancient backup of my HO1.4 databases. Let's see if I can read them in.

**Reply by tychobrailleur** [reply]
Out of curiosity, what is preventing from rebuilding the database on a new HO install by downloading all data from Hattrick, and potentially re-importing HRF files?   What would be missing from the new database that is in the old one?  This would sound like a safer approach to me, as we have seen quite a few H2 corruptions from migrations.

**Reply by tychobrailleur** [reply]
(IMHO it would be more valuable to be able to re-build the whole db at a push of a button, which would be cross-version – which we should already be able to do – rather than wasting time resurrecting old code and introduce more maintenance troubles for a very limited number of users – your time is too valuable for that)

**Reply by wsbrenk** [reply]
@tychobrailleur A lot of historical data will be lost, Especially the old training info could not be restored from hattrick itself. The old hrf files have also changed their format over the time. It is not sure that they can be imported anymore.

the HO4.2 patch i mentioned below is my favorite solution to catch the users of the famous HO1.4 version. There still seem to be a lot of them.

**Comment by masterpatje** [comment]
I think that is an excellent idea. More -old legend- users adopting the newer version could mean more new insights.

Please write down the steps when you try to do that so that specific instructions can be given for users.

I stayed quite some time on HO3 as main database, with recent developments last year (HO6! En HO7) that is no longer the case. I fully went over to use HO7.

Hartelijke groet,

Patrick van Wijnen

Op 12 mrt. 2023 om 10:49 heeft wsbrenk ***@***.***> het volgende geschreven:

﻿

I just did some research in the development history. The updateability of versions prior to 3 was removed in 2021. If the database API is still able to support the old databases today, I could try to rebuild these capabilities.
Since there still seem to be quite a few users using version 1.4, this might be a good idea. I just found an ancient backup of my HO1.4 databases. Let's see if I can read them in.

**Comment by wsbrenk** [comment]
@radpopl it is not as easy as i hoped when i started this morning, since the database engine was updated from version 1.8.0 in my 1.4 instance to 2.7.1 of HO7. Later is not able to access the old db files.  
up to 2 steps of database file upgrade are necessary, before HO can run its own updates.

1.8.0 -> To upgrade an old database created with version 1.8.x, you can use HSQLDB version 2.3.x to 2.5 to open the database and perform SHUTDOWN SCRIPT. You can then open the database with version 2.6 or later.

http://www.hsqldb.org/doc/2.0/guide/deployment-chapt.html#dec_upgrade_database

Could you please check the file database.properties, if your 1.4 database is of version 1.8 too?
version=1.8.0

**Reply by radpopl** [reply]
#HSQL Database Engine
#Sun Mar 12 00:17:00 CET 2023
hsqldb.script_format=0
runtime.gc_interval=0
sql.enforce_strict_size=false
hsqldb.cache_size_scale=8
readonly=false
hsqldb.nio_data_file=true
hsqldb.cache_scale=14
version=1.8.0
hsqldb.default_table_type=memory
hsqldb.cache_file_scale=1
hsqldb.log_size=200
modified=yes
hsqldb.cache_version=1.7.0
hsqldb.original_version=1.8.0
hsqldb.compatible_version=1.8.0

**Comment by wsbrenk** [comment]
@radpopl HO4.1 was the latest version running with hsqldb 2.4.1 which should be able to upgrade old HO1.4 databases (1.8.0).

Could you please try, to open a copy of your database with HO4.1. If that is not successful we could try to produce a HO4.2 patch that fix the errors that you will observe. Closing HO4 will produce a database instance that should be readable by HO7.

**Reply by radpopl** [reply]
I downloaded 4.1, unpacked. Despite Java installed, after running HO.exe I get:

No JVM could be found on your system.
Please define EXE4J_JAVA_HOME
to point to an installed 64-bit JDK orJRE or download a JRE
from www.java.com.

**Reply by radpopl** [reply]
I downloaded the version with JRE. After running HO.exe, the program appears in the Task Manager for a while, then disappears. Nothing is shown on the screen.

**Comment by wsbrenk** [comment]
hsqldb update|version|HO releases
-|-|-
01.11.2022|2.7.1|HO6.2, HO7
07.08.2022|2.7.0|HO6.1
23.07.2022|2.5.0|HO6.0
18.11.2021|2.6.1|HO5.0, HO5.1, HO5.2
07.10.2018|2.4.1|HO4.1, HO4.0, HO3.0, HO2.1, HO2.0, HO1.437, HO1.436, HO1.435

**Comment by masterpatje** [comment]
That one is not open on execution. You have to seperately open it after installation.

This was fixed in HO6 if I remember correctly.

Hartelijke groet,

Patrick

Op 12 mrt. 2023 om 14:34 heeft radpopl ***@***.***> het volgende geschreven:

﻿

I downloaded the version with JRE. After running HO.exe, the program appears in the Task Manager for a while, then disappears. Nothing is shown on the screen.

**Reply by radpopl** [reply]
So what should I do? I downloaded the installer version, but it also throws a Java missing error, also when pointing to the java.exe file.
The JRE version starts but closes after a while.

https://user-images.githubusercontent.com/11005341/224549389-7422d20e-4024-431e-b9dd-388523970838.mp4

**Reply by wsbrenk** [reply]
i just tried the following:

Download the zip file [HO-4.1.3310.2-portable-win-JRE.zip](https://github.com/akasolace/HO/releases/download/4.1/HO-4.1.3310.2-portable-win-JRE.zip)
unpack to ho4.1-portable
copied my HO1.4 database directory  to ho4.1-portable/db
started HO.exe (my HO1.4 database instance was an empty one;-(
after closing HO, the database should be updated.

**Comment by masterpatje** [comment]
I am on a mobile device so cant see the mp4
But you neer the JRE version. Install it, then it appears to close. If I am right you can see that HO is installed (via de windows menu) or if applicable a icon on the desktop.

When you click and open it, it should work. Otherwise it is due to another problem and you need help from wsbrenk.

Hartelijke groet,

Patrick

Op 12 mrt. 2023 om 14:58 heeft radpopl ***@***.***> het volgende geschreven:

﻿

So what should I do? I downloaded the installer version, but it also throws a Java missing error, also when pointing to the java.exe file.
The JRE version starts but closes after a while.

https://user-images.githubusercontent.com/11005341/224549389-7422d20e-4024-431e-b9dd-388523970838.mp4

**Reply by radpopl** [reply]
JRE is the same as java from java.com? The latter I have always had, because I use software that uses it - Hattrick Organizer 1.434 or Universal Media Player. HO3.0 started for me, 4.1 does not want to.

**Comment by masterpatje** [comment]
JE = java realtime engine
I could not make it work without

Hartelijke groet,

Patrick

Op 12 mrt. 2023 om 15:16 heeft radpopl ***@***.***> het volgende geschreven:

﻿

JRE is the same as java from java.com? The latter I have always had, because I use software that uses it - Hattrick Organizer 1.434 or Universal Media Player. HO3.0 started for me, 4.1 does not want to.
